service/history/execution/mutable_state_builder.go

```diff
@@ -1360,6 +1363,7 @@ func (e *mutableStateBuilder) DeleteActivity(
 		e.logDataInconsistency()
 	}
 
+	delete(e.updateActivityInfos, scheduleEventID)
```

Comment:
Assuming DeleteActivity should always be called after UpdateActivity (otherwise it would be a bug).